doc/source/whatsnew/v0.24.0.txt

```diff
@@ -623,7 +623,7 @@ MultiIndex
 ^^^^^^^^^^
 
 - Removed compatibility for MultiIndex pickles prior to version 0.8.0; compatibility with MultiIndex pickles from version 0.13 forward is maintained (:issue:`21654`)
--
+- Fix issue when creating :class:`MultiIndex` from mixed type arrays including tuples (:issue:'15457'). In some cases df.set_index(['A','B']) raised a ``TypeError unorderable types`` when trying to set multiple columns as indices.
```

Comment:
We don't create the MultiIndex "from mixed type arrays", and I think the two sentences can be merged.

The following should work: "Fix ``TypeError`` in Python 3 when creating ``MultiIndex`` in which some labels are tuples"

Use backticks (not single quotes) around the issue number.

Reply:
pls add the issue number

Reply:
The problem seems to occur as well when the column types I want to index are of type object. Not sure why this is related. In that case they are unorderable (I think), but it should still be possible to set them as index I assume? I'll try to create a test for that as well.

Reply:
> The problem seems to occur as well when the column types I want to index are of type object.

@dickreuter Not sure I follow you: when "_some labels are tuples_", then the column types are _necessarily_ of type ``object``. That's the only dtype that can hold tuples.

> pls add the issue number

@jreback it's already there

Reply:
@toobaz Yes so the description should make reference to the object type rather than the tuple, because the former is the actual cause for the error.

Reply:
> the description should make reference to the object type rather than the tuple, because the former is the actual cause for the error.

Not sure what makes you think this, but
 
```
problem = (0, 1)
pd.DataFrame([[2, 1], [4, problem]]).set_index([0, 1])
```
works just fine if you replace the first line with ``problem = 'a'``.

(The ultimate reason why tuples are a problem is #22080.)

Reply:
But if you have objects (I had ctype items) in the rows of column 'a', then it will fail. That's what I mean. It's not just the column name that matters.